client/api/src/main/java/io/smallrye/graphql/client/typesafe/api/TypesafeClientResponse.java

```diff
@@ -0,0 +1,91 @@
+package io.smallrye.graphql.client.typesafe.api;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import jakarta.json.JsonObject;
+
+import io.smallrye.graphql.client.GraphQLError;
+
+public final class TypesafeClientResponse<T> extends ErrorOr<T> {
```

Comment:
> Maybe `GraphQLResponse` would be a better class name?

I would agree with renaming to `TypesafeResponse`, as we need to clearly distinguish it from `io.smallrye.graphql.client.Response` which is used by dynamic clients


> Just a thought: It might be useful to be able to return instances of this class also on the server-side. Analogous to the JAX-RS `Response` class.

What would be the usage? The application can add HTTP headers to the response (via a web filter in WildFly or `CurrentVertxRequest` in Quarkus) and extensions too (via an injected `SmallRyeContext`), so it looks unnecessary. We would also have to add `ErrorOr` to the server and then applications would have to manually instantiate `GraphQLError`s instead of simply throwing exceptions that get mapped to them automatically.

Reply:
You're right.